### PR TITLE
use Rust 1.82 build image

### PR DESCRIPTION
**Pull request by fbjork**
This also updates the docker image to use Debian bookworm.

Closes: GB-7807

### CODE REVIEW

File: gateway/Dockerfile [hunk 1]
```diff
@@ -35,13 +28,12 @@ COPY ./gqlint ./gqlint
...
 
-RUN cargo build --release --target "$(rustc -vV | sed -n "s|host: ||p")" --bin grafbase-gateway &&\
-    mv "/grafbase/target/$(rustc -vV | sed -n "s|host: ||p")/release/grafbase-gateway" /grafbase/target/release/grafbase-gateway
+RUN cargo build --release --bin grafbase-gateway
 
 #
 # === Final image ===
 #
-FROM alpine:3.20
+FROM debian:bookworm
```
Comment (obmarg) [inline gateway/Dockerfile:36]: Not that fussed if no, but could we use `bookworm-slim` as the base of the final image?

File: gateway/Dockerfile [hunk 2]
```diff
@@ -53,9 +45,9 @@ LABEL org.opencontainers.image.url='https://grafbase.com' \
 WORKDIR /grafbase
 
 # used curl to run a health check query against the server in a docker-compose file
-RUN apk add --no-cache curl
+RUN apt update && apt install -y curl
```
Comment (obmarg) [inline gateway/Dockerfile:48]: Not sure how much difference it makes, but I remember in previous job we used to delete apt stuff in the same line as `apt update` to keep layer size down.

```suggestion
RUN apt update && apt install -y curl && rm -rf /var/lib/apt/lists/*
```

File: gateway/Dockerfile (hunk 1 above)
Comment (fbjork) [reply gateway/Dockerfile:36]: do we want to use a specific version?

File: gateway/Dockerfile [hunk 3]
```diff
@@ -1,17 +1,10 @@
 #
 # === Build image ===
 #
-FROM rust:1.80-alpine3.20 AS chef
-
-# Patching musl to use mimalloc, see docker/README.md.
-# Source: https://github.com/tweag/rust-alpine-mimalloc
-COPY docker/build.sh docker/mimalloc.diff /tmp/
```
Comment (fbjork) [inline gateway/Dockerfile:8]: can we remove the docker folder now?

File: gateway/Dockerfile [hunk 4]
```diff
@@ -1,17 +1,10 @@
...
-RUN /tmp/build.sh
-ENV LD_PRELOAD=/usr/lib/libmimalloc.so
-ENV RUSTFLAGS="-C target-feature=+crt-static"
+FROM rust:1.82-bookworm AS chef
 
 COPY rust-toolchain.toml rust-toolchain.toml
-RUN apk add --no-cache musl-dev && cargo install --target "$(rustc -vV | sed -n "s|host: ||p")" cargo-chef
+RUN cargo install --target "$(rustc -vV | sed -n "s|host: ||p")" cargo-chef
```
Comment (obmarg) [inline gateway/Dockerfile:7]: Is this `--target` even needed now?  Would have thought not.

```suggestion
RUN cargo install cargo-chef
```

**Review by obmarg — APPROVED** [review]

File: gateway/Dockerfile (hunk 1 above)
Comment (pimeys) [reply gateway/Dockerfile]: Well the build image doesn't have latest debian, and points to bookworm, so we do too. This one will be supported for a _loooong_ time.

**Pull request merged**